### PR TITLE
dd: interpret Stdin as File to seek it

### PR DESCRIPTION
**Pull request by ArtemSkrebkov**
It fixes a test including `dd` from `tests/misc/head-c.sh`.
Corresponding issue: https://github.com/uutils/coreutils/issues/3008#

I expected that `tests/dd/not-rewound.sh` would also be fixed, but unfortunately, it is not the case. It seems that https://github.com/uutils/coreutils/issues/3626 is still relevant.

#### Background

An attempt to skip bytes by `read_skip` leads to consuming all stdin by a rust application. Another approach to tackle this is to interpret `stdin` as `File`. More details in https://github.com/uutils/coreutils/issues/3008#issuecomment-1136024931

### CODE REVIEW

**Comment by sylvestre** [comment]
Could you please add a test?
We have plenty of tests in `tests/by-util/test_*`

File: src/uu/dd/src/dd.rs [hunk 1]
```diff
@@ -71,17 +71,23 @@ impl Input<io::Stdin> {
             iflags,
         };
 
-        // The --skip and --iseek flags are additive. On a stream, they discard bytes.
+        // there is way to seek stdin, so we interpret stdin as a File
+        // to make it seekable
+        #[cfg(any(target_family = "unix", target_family = "wasi"))]
+        let mut seekable_src = unsafe {
+            use std::os::unix::io::{AsRawFd, FromRawFd};
+            std::fs::File::from_raw_fd(std::io::stdin().as_raw_fd())
```
Comment (the8472) [inline src/uu/dd/src/dd.rs:79]: (driveby comment, I don't have much context) Unless `seekable_src` is fed to `mem::forget` at some point this likely is a bad idea because `File::drop` will then close the stdin file descriptor. You should either `try_clone` (dup) it or wrap this in a `ManuallyDrop`.

File: tests/by-util/test_dd.rs [hunk 2]
```diff
@@ -1186,3 +1186,16 @@ fn test_final_stats_si_iec() {
     let s = result.stderr_str();
     assert!(s.starts_with("2+0 records in\n2+0 records out\n1024 bytes (1 KB, 1024 B) copied,"));
 }
+
+#[test]
+fn test_partial_skip_stdin() {
```
Comment (ArtemSkrebkov) [inline tests/by-util/test_dd.rs:1191]: @sylvestre 

Well... I added the test, but it does not work as expected and fails with `Illegal seek`. As any other test which can try seek `stdin`.

However, it can work outside the test system (for example, if I run `tests/misc/head-c.sh`)

I am confused why it does not work in the test system. Let me know if there are any insights on this.

File: src/uu/dd/src/dd.rs (hunk 1 above)
Comment (ArtemSkrebkov) [reply src/uu/dd/src/dd.rs:79]: That makes sense. Will fix it once I figure out why the current solution works if I use `dd` in command line and does not work if I run tests from `tests/by-utils`.

**Comment by sylvestre** [comment]
Could you please fix that issue? thanks

```

error[E0425]: cannot find value `i` in this scope
  --> src\uu\dd\src/dd.rs:93:12
   |
93 |         Ok(i)
   |            ^ not found in this scope
```

**Comment by ArtemSkrebkov** [comment]
> Could you please fix that issue? thanks
> 
> ```
> 
> error[E0425]: cannot find value `i` in this scope
>   --> src\uu\dd\src/dd.rs:93:12
>    |
> 93 |         Ok(i)
>    |            ^ not found in this scope
> ```

Fixed.

**Comment by sylvestre** [comment]
a bunch of tests are broken, could you please have a look? thanks

**Comment by ArtemSkrebkov** [comment]
Yeah. Right. I described why the tests fail in the previous comment. I will try to debug it more, but if there is someone, who is more proficient in the project, I can chat with, that would probably help to fix the issue faster.

**Review by sylvestre — CHANGES_REQUESTED** [review]
fails to build on a bunch of jobs

**Comment by sylvestre** [comment]
@ArtemSkrebkov ping?

**Comment by ArtemSkrebkov** [comment]
No luck since the last debugging attempt.
I believe, we probably can close until better times.

**Comment by tertsdiepraam** [comment]
Closing until better times :)
